### PR TITLE
POC: Make subproject sync interval configurable #8299

### PR DESCRIPTION
**Pull request by vbradnitski**

### CODE REVIEW

File: modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java [hunk 1]
```diff
@@ -30,10 +30,10 @@
 import com.enonic.xp.security.auth.AuthenticationInfo;
 
 @Component(immediate = true)
-public class ProjectNodeEventListener
+public class ProjectContentEventListener
     implements EventListener
 {
-    private final static Logger LOG = LoggerFactory.getLogger( ProjectNodeEventListener.class );
+    private final static Logger LOG = LoggerFactory.getLogger( ProjectContentEventListener.class );
```
Comment (jsi) [inline modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java:36]: Codacy found an issue: ['static' modifier out of order with the JLS suggestions.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6117594)

File: modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectEventListener.java [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+import com.enonic.xp.security.PrincipalKey;
+import com.enonic.xp.security.RoleKeys;
+import com.enonic.xp.security.User;
+import com.enonic.xp.security.auth.AuthenticationInfo;
+
+@Component(immediate = true)
+public class ProjectEventListener
+    implements EventListener
+{
+    private final static Logger LOG = LoggerFactory.getLogger( ProjectEventListener.class );
```
Comment (jsi) [inline modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectEventListener.java:33]: Codacy found an issue: ['static' modifier out of order with the JLS suggestions.](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6117594)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectEventListener.java  4
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentProjectSyncActivator.java  2
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectNotFoundException.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentProjectSyncTask.java  1
- modules/core/core-api/src/main/java/com/enonic/xp/project/ParentProjectSynchronizer.java  4
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java  4
         

Clones added
============
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectEventListener.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6117594&bid=20019677)

File: modules/runtime/src/home/config/com.enonic.xp.content.cfg [hunk 3]
```diff
@@ -1 +1,2 @@
-#auditlog.enabled = true
\ No newline at end of file
+#auditlog.enabled = true
+#content.sync.period = 1000
```
Comment (rymsha) [inline modules/runtime/src/home/config/com.enonic.xp.content.cfg:2]: Please use Duration format instead.

**Review by rymsha — CHANGES_REQUESTED** [review]

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectEventListener.java  4
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectNotFoundException.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentProjectSyncTask.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java  4
- modules/core/core-api/src/main/java/com/enonic/xp/project/ParentProjectSynchronizer.java  4
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ParentProjectSyncActivator.java  2
         

Clones added
============
- modules/core/core-project/src/main/java/com/enonic/xp/core/impl/project/ProjectEventListener.java  1
- modules/core/core-content/src/main/java/com/enonic/xp/core/impl/content/ProjectContentEventListener.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=6117594&bid=20019677)

**Review by rymsha — APPROVED** [review]

**Pull request merged**